server/src/com/thoughtworks/go/server/service/plugins/processor/SessionDAORequestProcessor.java

```diff
@@ -0,0 +1,96 @@
+/*************************GO-LICENSE-START*********************************
+ * Copyright 2014 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *************************GO-LICENSE-END***********************************/
+
+package com.thoughtworks.go.server.service.plugins.processor;
+
+import com.google.gson.Gson;
+import com.thoughtworks.go.plugin.api.request.GoApiRequest;
+import com.thoughtworks.go.plugin.api.response.DefaultGoApiResponse;
+import com.thoughtworks.go.plugin.api.response.GoApiResponse;
+import com.thoughtworks.go.plugin.infra.DefaultGoApplicationAccessor;
+import com.thoughtworks.go.plugin.infra.GoPluginApiRequestProcessor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import javax.servlet.http.HttpSession;
+import java.util.Map;
+
+@Component
+public class SessionDAORequestProcessor implements GoPluginApiRequestProcessor {
+    private static final String STORE_IN_SESSION = "store-in-session";
+    private static final String GET_FROM_SESSION = "get-from-session";
+    private static final String REMOVE_FROM_SESSION = "remove-from-session";
+
+    @Autowired
+    public SessionDAORequestProcessor(DefaultGoApplicationAccessor goApplicationAccessor) {
+        goApplicationAccessor.registerProcessorFor(STORE_IN_SESSION, this);
```

Comment:
right. this is to store state across web requests like [1](https://github.com/gocd/go-plugins/pull/14/files#diff-8285148299afe93bde971ae9ebb96a67R108) & [2](https://github.com/gocd/go-plugins/pull/14/files#diff-8285148299afe93bde971ae9ebb96a67R121). Plugin couldn't have written to file etc. since the right file needs to be mapped back for same session.